modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/Marshaling.cs

```diff
@@ -0,0 +1,1323 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+// ReSharper disable InconsistentNaming
+
+namespace Godot.NativeInterop
+{
+    // We want to use full name qualifiers here even if redundant for clarity
+    [SuppressMessage("ReSharper", "RedundantNameQualifier")]
+    internal static class Marshaling
+    {
+        public static unsafe void SetFieldValue(FieldInfo fieldInfo, object obj, godot_variant* value)
+        {
+            var valueObj = variant_to_mono_object_of_type(value, fieldInfo.FieldType);
+            fieldInfo.SetValue(obj, valueObj);
+        }
+
+        public static Variant.Type managed_to_variant_type(Type type, ref bool r_nil_is_variant)
+        {
+            switch (Type.GetTypeCode(type))
+            {
+                case TypeCode.Boolean:
+                    return Variant.Type.Bool;
+                case TypeCode.Char:
+                    return Variant.Type.Int;
+                case TypeCode.SByte:
+                    return Variant.Type.Int;
+                case TypeCode.Int16:
+                    return Variant.Type.Int;
+                case TypeCode.Int32:
+                    return Variant.Type.Int;
+                case TypeCode.Int64:
+                    return Variant.Type.Int;
+                case TypeCode.Byte:
+                    return Variant.Type.Int;
+                case TypeCode.UInt16:
+                    return Variant.Type.Int;
+                case TypeCode.UInt32:
+                    return Variant.Type.Int;
+                case TypeCode.UInt64:
+                    return Variant.Type.Int;
+                case TypeCode.Single:
+                    return Variant.Type.Float;
+                case TypeCode.Double:
+                    return Variant.Type.Float;
+                case TypeCode.String:
+                    return Variant.Type.String;
+                default:
+                {
+                    if (type == typeof(Vector2))
+                        return Variant.Type.Vector2;
```

Comment:
VS Code's C# formatter indents these blocks further (starting with the `{`). I don't really have an opinion on this, but I'm mentioning it in case we wanted to change this to avoid fighting with tools.

Reply:
This is the `.editorconfig` rule you are looking for:
```
csharp_indent_case_contents_when_block = false
```

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/formatting-rules#csharp_indent_case_contents_when_block